### PR TITLE
#101-2 func as iterator

### PR DESCRIPTION
**Pull request by Timmeey**
Step two towards #101
Based on #162

### CODE REVIEW

**Comment by 0crat** [comment]
@timmeey I'm not managing this repo, remove the [webhook](https://github.com/yegor256/cactoos/settings/hooks) or contact me in [Slack](http://www.zerocracy.com) //cc @yegor256

File: src/main/java/org/cactoos/list/FuncAsIterator.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+/**
+ * Iterator over the output of a function.
+ *
+ * @author Tim Hinkes (timmeey@timmeey.de)
+ * @version $Id$
+ * @param <X> The input type for the Function
+ * @param <Y> The output Type of the Function
+ * @since 0.7
+ */
+public final class FuncAsIterator<X, Y> implements Iterator<Y> {
```
Comment (yegor256) [inline src/main/java/org/cactoos/list/FuncAsIterator.java]: @Timmeey how is it different from `TransformedIterator`?

**Comment by yegor256** [comment]
@Timmeey see my comment above

File: src/main/java/org/cactoos/list/FuncAsIterator.java (hunk 1 above)
Comment (Timmeey) [inline src/main/java/org/cactoos/list/FuncAsIterator.java]: it seems to be called `MappedIterator` now, but yeah. good question.

**Comment by 0crat** [comment]
Oops! Job `gh:yegor256/cactoos#163` was not in scope

**Pull request closed without merging**